keps/sig-api-machinery/4006-transition-spdy-to-websockets/kep.yaml

```diff
@@ -0,0 +1,43 @@
+title: Transition from SPDY to Websockets
+kep-number: 4006
+authors:
+  - "@seans3"
+owning-sig: sig-api-machinery
+participating-sigs:
+  - sig-cli
+status: provisional
+# status other states: # |implementable|implemented|deferred|rejected|withdrawn|replaced
+creation-date: 2023-05-15
+reviewers:
+  - "@aojea"
+  - "@ardaguclu"
+approvers:
+  - "@deads2k"
+  - "@jpbetz"
+
+# The target maturity stage in the current dev cycle for this KEP.
+stage: alpha
+
+# The most recent milestone for which work toward delivery of this KEP has been
+# done. This can be the current (upcoming) milestone, if it is being actively
+# worked on.
+latest-milestone: "v1.28"
+
+# The milestone at which this feature was, or is targeted to be, at each stage.
+milestone:
+  alpha: "v1.28"
+  beta: "v1.29"
+  stable: "v1.30"
+
+# The following PRR answers are required at alpha release
+# List the feature gate name and the components for which it must be enabled
+feature-gates:
+  - name: Websockets
+    components:
```

Comment:
@aojea @ardaguclu @jpbetz Hoping to get feedback on the components requiring a feature flag. 

1. Do we need feature flags for both the API Server **and** `kubectl` ?
2. Can we get away with just a feature flag on `kubectl` ? The code paths in the API Server (`StreamTranslatorProxy`) should not be reachable without the new `kubectl` requesting v5 of `RemoteCommand` subprotocol.
3. If we have feature flags for both components, can they be the same name ?

Reply:
I think, `v5.channel.k8s.io` subprotocol is already a some kind of a feature flag in API server. Everything will work as is, as long as clients send requests in current subprotocols(v4..v1) and `v5.channel.k8s.io` subprotocol will be used in kubectl, only when it's feature flag is enabled.

Reply:
summoning @liggitt , he is a natural for these things ;)

Reply:
Added feature flags for both `kubectl` and the API Server for this alpha KEP. If we change it, I will update the KEP.